src/esp/sensor/Sensor.h

```diff
@@ -114,8 +114,14 @@ class Sensor : public Magnum::SceneGraph::AbstractFeature3D {
 
   virtual bool isVisualSensor() { return false; }
 
-  // visual sensor should implement and override this function
-  virtual void setProjectionMatrix(gfx::RenderCamera& targetCamera){};
+  // visual sensor should implement and override the following functions
+  virtual Sensor& setProjectionMatrix(gfx::RenderCamera& targetCamera) {
```

Comment:
I think it would be better to have the sensor provide all these things via getters rather than giving them the camera and having them set it.  That way people don't have to be concerned with the specifics of how to set the projection matrix and just need to provide it.

Reply:
Such a getter function will return a matrix (e.g., a 4x4 matrix), which is an intermediate value passing to the "render camera". Certainly we can have such getter functions. And I can see such functions can be very useful in some cases. 
But if it is better, I doubt.
I prefer the way used here:
```sensor.setProjectMatrixTo(camera);```
A user treats the function as a "black box" and she does not need to be concerned with the implementation details underlying.

Reply:
It depends on what you think the user wants to do.  Does the user want to be able to set the set the projection matrix on a new camera, or does the user want to be able to add a new sensor?  This API makes sense for the former, doing this with a getter makes sense for the latter.

Reply:
@erikwijmans : Yes, certainly we can add new getter functions as you suggested.